src/components/FAB.js

```diff
@@ -0,0 +1,60 @@
+import React, {memo} from 'react';
+import {Pressable, Animated} from 'react-native';
+import PropTypes from 'prop-types';
+import {PlusIcon} from './Expensicons';
+import styles from '../styles/styles';
+import colors from '../styles/colors';
+import themeColors from '../styles/themes/default';
+
+const AnimatedPlusIcon = Animated.createAnimatedComponent(PlusIcon);
+const AnimatedPressable = Animated.createAnimatedComponent(Pressable);
+
+const propTypes = {
+    // Callback to fire on request to toggle the FAB
+    onPress: PropTypes.func.isRequired,
+
+    // Animated value to control the animation
+    animatedValue: PropTypes.instanceOf(Animated.Value).isRequired,
```

Comment:
I think passing in the `animatedValue` is more of an implementation detail that should be the responsibility of this component and it would be better to just pass `isActive`. One idea to achieve this would be:

1. Switch to a class component and keep the `Animated.Value` here
2. Check to see if the `isActive` prop is changing from `true/false` in `componentDidUpdate`
3. Update the `Animated.Value` accordingly

What do you think?

Reply:
This is an important one.
I am going to try to alter it and comment again with something soon.

Reply:
Just realized my comment above could be seen as like the other reviews were not important. Sorry about that just meant this one is larger / not as small.

Finished redesigning the code to fit the comment. I should have done this way from the beginning, but the solution didn't cross my mind.

The new commit has this changes.